tests/queries/0_stateless/01600_remerge_sort_lowered_memory_bytes_ratio.sql

```diff
@@ -0,0 +1,7 @@
+-- Check remerge_sort_lowered_memory_bytes_ratio setting
```

Comment:
This test proves that the settings works and should not be set too large.
But is there an example of the benefit from changing this setting?

Reply:
>This test proves that the settings works and should not be set too large.

Yeah, that was a debug value

>But is there an example of the benefit from changing this setting?

Memory usage will be smaller or are you talking about the case when default ratio 2 is not good enough?

Reply:
When the default is not good enough?

Reply:
>When the default is not good enough?

What I got with real data is:

```
MergeSortingTransform: Memory usage is lowered from 2.20 GiB to 1.17 GiB`
```

So 2 was slightly not enough (1.8 will be ok), I've updated the test and now it fails for remerge_sort_lowered_memory_bytes_ratio=2 (default), but the data is still synthetic.

Reply:
Also now it will have more messages in log, so that it is easy to figure out how this can be fixed

Reply:
Frankly speaking I'm not sure about introducing one more setting for this, it will be better if this will be done automatically, and instead of disabling remerge completely, making it less frequent should work (once memory does not reduced 2x, increase number of rows to remerge, i.e. not limit*2, but limit*3/4 and so on)

Reply:
Actually I have a patch that will improve remerge logic, and there is possible two options:
- increase remerge limit to LIMIT
- increase remerge limit to max_block_size

The first option may still increase memory usage, while the second may make things slower.

So maybe setting makes sense, for example in case of low memory setup or low memory limit for query (and also for tests), and on top of it the remerge logic can be improved

Reply:
@alexey-milovidov what do you thing?